keps/sig-autoscaling/draft-20181106-in-place-update-of-pod-resources.md

```diff
@@ -0,0 +1,170 @@
+---
+kep-number: draft-20181106
+title: In-place Update of Pod Resources
+authors:
+  - "@kgolab"
+  - "@bskiba"
+  - "@schylek"
+owning-sig: sig-autoscaling
+participating-sigs:
+  - sig-node
+  - sig-scheduling
+reviewers:
+  - "@bsalamat"
+  - "@derekwaynecarr"
+  - "@dchen1107"
+approvers:
+  - TBD
+editor: TBD
+creation-date: 2018-11-06
+last-updated: 2018-11-06
+status: provisional
+see-also:
+replaces:
+superseded-by:
+---
+
+# In-place Update of Pod Resources
+
+## Table of Contents
+
+   * [In-place Update of Pod Resources](#in-place-update-of-pod-resources)
+      * [Table of Contents](#table-of-contents)
+      * [Summary](#summary)
+      * [Motivation](#motivation)
+         * [Goals](#goals)
+         * [Non-Goals](#non-goals)
+      * [Proposal](#proposal)
+         * [API Changes](#api-changes)
+         * [Flow Control](#flow-control)
+         * [Notes](#notes)
+         * [Risks and Mitigations](#risks-and-mitigations)
+      * [Graduation Criteria](#graduation-criteria)
+      * [Implementation History](#implementation-history)
+      * [Alternatives](#alternatives)
+
+## Summary
+
+This proposal aims at allowing Pod resource requests & limits to be updated
```

Comment:
We have many many use cases that needs changing other fields in Pod (e.g. environment vars). Could we also consider in-place update for them if it is possible?

Reply:
@hex108 We want to keep the scope of this KEP limited to container CPU and memory resources which falls within the definition of vertical scaling. Do you have CPU/memory scaling use-cases that you feel might not be addressed by this KEP?

Reply:
@vinaykul Thanks! I get it now. I'll comment on other lines for our use cases.

Reply:
I second @vinaykul in keeping the KEP limited to container resources only.

@hex108 , I imagine you'd need to update whole PodSpec when updating environment variables (since they can be used in PodSpec) and to me this looks like a full Pod restart which is doable already. Or do you have use-cases for updated environment variables w/o Pod/container restarts?